lib/go-tc/parameters.go

```diff
@@ -255,6 +255,41 @@ type ProfileParametersNullableResponse struct {
 	Response []ProfileParametersNullable `json:"response"`
 }
 
+// ProfileParameterV40 is a relationship between a Profile and some Parameter
+// assigned to it as it appears in version 4.0 of the Traffic Ops API's
+// responses to the /profileparameters endpoint.
+type ProfileParameterV40 struct {
```

Comment:
Without `V40` this collides with a [pre-existing `ProfileParameter` structure](https://pkg.go.dev/github.com/apache/trafficcontrol@v5.1.1+incompatible/lib/go-tc#ProfileParameter)

Reply:
`ProfileParam`?

Reply:
Idk man

"What type does the `GetProfileParameters` method return again? That uses the `/profileparameter` endpoint, right?"

"I think it returns a `ProfileParameter` array"

"No wait, isn't it a `ProfileParameterNullable` array?"

"You're all wrong, it uses `/profileparameters` and returns an array of `ProfileParam`s"

Idk how much "V4" really helps, I guess. I just wish we could do something about all that...

Reply:
I know, it is a massive mess. I was thinking about the possibility of just creating a new package with "clean" structs and names to kind of start from scratch, but that would be a much broader change to also the TO internals.

Part of me (most of me) thinks that preserving struct names is just a massive waste of effort and we should be more free to make breaking changes since the amount of things using the Go client are mostly within ATC itself, and external things are probably largely relying on type inference anyways, making them immune to type name changes.

Reply:
I'm really hoping that now that we're using Go modules, in time - and especially with ORT splitting off into its own repo, possibly - we can make breaking changes to _anything_ with major version changes.